vent/service/consumer.go

```diff
@@ -321,10 +315,21 @@ func decodeEvent(eventName string, header *exec.Header, log *exec.LogEvent, abiS
 
 	// for each decoded item value, stores it in given item name
 	for i, input := range eventAbiSpec.Inputs {
-    
-		data[input.Name] = unpackedData[i]
-    
-		l.Debug("msg", fmt.Sprintf("Unpacked data items: unpackedData[%v] = %v, input.Name = %v", i, unpackedData[i], input.Name), "eventName", eventName)
+
+		switch v := unpackedData[i].(type) {
+		case *crypto.Address:
+			data[input.Name] = v.Bytes()
+		case *big.Int:
+			if v.IsInt64() {
```

Comment:
If I'm not wrong these `IsInt64` and `IsUint64` methods only check if the numbers wrapped by the `big.Int` type can be represented as an `int64` and `uint64` respectively. So I'm not sure if they are useful for doing a proper type check here. For example, a positive value will return true for both methods (check this playground for reference: https://play.golang.org/p/z6Fl18fQIPP), and if the number is bigger than the size of an `int64` it will go through the `else` and `v.Uint64()` will return an undefined value as it says here https://golang.org/pkg/math/big/#Int.Uint64.

Reply:
you're right i will change that to `v.String()` so `postgres` will implicitly convert that to `numeric`